### PR TITLE
Change the way that max premium is calculated

### PR DESCRIPTION
**Pull request by MilGard91**
## Context

Issue: #79 

## Changes proposed in this pull request

- Add a function that will calculate premium for next 4 tranches and return max premium of the pool

## Test plan

Please describe the tests cases that you ran to verify your changes. Add further instructions on 
how to run them if needed (i.e. migration / deployment scripts, env vars, etc).

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: src/lib/capacityEngine.js [hunk 1]
```diff
@@ -77,9 +120,7 @@ function capacityEngine(store, productIds, time, period = 30) {
         // the maximum price a user would get can only be determined if the entire available
         // capacity is bought because the routing will always pick the cheapest
         // so we're summing up the premium for all pools and then calculate the average at the end
-        const poolPremium = product.useFixedPrice
-          ? calculateFixedPricePremiumPerYear(availableInNXM, basePrice)
-          : calculatePremiumPerYear(availableInNXM, basePrice, usedInNxm, totalInNXM);
+        const poolPremium = calculateMaxPremium(product, pool, basePrice, time);
```
Comment (roxdanila) [inline src/lib/capacityEngine.js:123]: fixed price products don't have surge, so it's fine to not enter to for loop and return here, as before

File: src/lib/capacityEngine.js [hunk 2]
```diff
@@ -1,12 +1,55 @@
...
+    // convert to nxm
+    const totalInNXM = total.mul(NXM_PER_ALLOCATION_UNIT);
+    const usedInNxm = used.mul(NXM_PER_ALLOCATION_UNIT);
+    const availableInNXM = availableCapacity.mul(NXM_PER_ALLOCATION_UNIT);
+
+    const poolPremium = product.useFixedPrice
+      ? calculateFixedPricePremiumPerYear(availableInNXM, basePrice)
+      : calculatePremiumPerYear(availableInNXM, basePrice, usedInNxm, totalInNXM);
+
+    maxPremium = bnMax(poolPremium, maxPremium);
```
Comment (roxdanila) [inline src/lib/capacityEngine.js:47]: you're adding up max premiums on different pools that could be on different periods. you have to first calculate maxes on all periods, sum them up on all pools, and only then find the absolut max

**Review by roxdanila — CHANGES_REQUESTED** [review]
see comments

**Comment by MilGard91** [comment]
Fixed the max premium calculation by extracting the productData reducer and using it to calculate the product data for next five tranches (4 - tranches 364 days, and max premium is 365 so just in case added one more iteration). Finding the max premium while iterating and also the capacity data for the requested period. All of this is skipped if the product has fixed price and calculating everything just for a requested period.

**Review by roxdanila — APPROVED** [review]

**Pull request merged**